java/ql/src/semmle/code/java/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -673,14 +673,15 @@ module Private {
       override predicate required(SummaryComponent c) { interpretSpec(_, _, c, this) }
     }
 
-    private class SummarizedCallableExternal extends SummarizedCallable {
-      SummarizedCallableExternal() { summaryElement(this, _, _, _) }
+    class SummarizedCallableExternal extends SummarizedCallable {
```

Comment:
Making use of `SummarizedCallableExternal` is potentially problematic as that means we only generate tests for those rows that translate without problems. It there's a typo in a signature, or method name, or anything like that then we don't generate any test case, so we fail to catch the error.

Reply:
I think, for a csv row with a package name matching something we specify when running the generator, we'll want to be able to generate test cases for csv rows that are wrong in ways that cause the test case to not compile.

Reply:
The invalid-row problem is addressed by the Python component, which introduces

```
query string getAFailedRow() {
  result = any(GenRow row | not exists(RowTestSnippet r | exists(r.getATestSnippetForRow(row))) | row)
}
```

This spots rows that didn't lead to any test generation, e.g. because a SummarizedCallableExternal wasn't matched, or a SummaryComponentStack parser didn't match.

Reply:
Gotcha. This code snippet could move into the qll file.